File: src/dataset/cpp/secret-handshake_exercism-tqa236/openAI/ChatGPT4_secret_handshake_v2.cpp
#include "secret_handshake.h"
#include <array>
#include <vector>
#include <string>

namespace secret_handshake {
constexpr std::array<const char *, 4> actions{"wink", "double blink",
                                              "close your eyes", "jump"};

std::vector<std::string> commands(unsigned int code) {
  std::vector<std::string> selected_actions;
  selected_actions.reserve(actions.size()); // Reserve memory to avoid reallocations
  for (size_t i = 0; i < actions.size(); ++i) {
    if (code & (1u << i)) selected_actions.emplace_back(actions[i]);
  }
  if (code & (1u << actions.size())) {
    std::reverse(selected_actions.begin(), selected_actions.end());
  }
  return selected_actions;
}
}  // namespace secret_handshake

File: src/dataset/typescript/secret-handshake_exercism-wackerow/openAI/ChatGPT4_secret-handshake_v2.ts
export default class {
  private readonly _steps = ['wink', 'double blink', 'close your eyes', 'jump'];

  constructor(private _secret: number) {}

  public commands(): string[] {
    const steps: string[] = this._steps.filter((_, i) => (this._secret >> i) & 1);
    return (this._secret & 16) ? steps.reverse() : steps;
  }
}